src/app/clusters/service-area-server/service-area-cluster-objects.h

```diff
@@ -238,19 +263,22 @@ struct AreaStructureWrapper : public chip::app::Clusters::ServiceArea::Structs::
             }
         }
 
-        if (areaDesc.landmarkTag != aOther.areaDesc.landmarkTag)
+        if (areaDesc.landmarkInfo.IsNull() != aOther.areaDesc.landmarkInfo.IsNull())
         {
             return false;
         }
 
-        if (areaDesc.positionTag != aOther.areaDesc.positionTag)
+        if (!areaDesc.landmarkInfo.IsNull())
         {
-            return false;
-        }
+            if (areaDesc.landmarkInfo.Value().landmarkTag != aOther.areaDesc.landmarkInfo.Value().landmarkTag)
```

Comment:
This is just comparing two optional structs, right? can you not just use the default comparison operator? 
ex
if (areaDesc.landmarkInfo != aOther.areaDesc.landmarkInfo) {
return false;
}

Reply:
same q for above.

Reply:
@cecille Tracked using https://github.com/project-chip/connectedhomeip/issues/34859. We'll address in a follow-up PR.

Reply:
@cecille , does a default comparison operator exist for such structs? This would have to be in the generated code and I haven't seen any implemented.